packages/frontend/src/pages/Send/useL1CanonicalBridge.ts

```diff
@@ -0,0 +1,136 @@
+import { useEffect, useState } from 'react'
+import { ChainId, NetworkSlug, Token } from '@hop-protocol/sdk'
+import { BigNumber, constants } from 'ethers'
+import Network from 'src/models/Network'
+import { useWeb3Context } from 'src/contexts/Web3Context'
+import logger from 'src/logger'
+import { findNetworkById, formatError, toTokenDisplay } from 'src/utils'
+import CanonicalBridge from 'src/models/CanonicalBridge'
+import { handleTransaction } from './useSendTransaction'
+import { useTransactionReplacement } from 'src/hooks'
+
+async function needsApproval(l1CanonicalBridge: CanonicalBridge, sourceToken, sourceTokenAmount) {
```

Comment:
the first parameter has a type but the other two do not. I think they should all have types